src/main/java/seedu/uninurse/logic/commands/AddTaskCommand.java

```diff
@@ -56,6 +56,11 @@ public CommandResult execute(Model model) throws CommandException {
 
         Patient personToEdit = lastShownList.get(index.getZeroBased());
 
+
+        if (personToEdit.getTasks().hasTask(task)) {
+            throw new CommandException(Messages.MESSAGE_DUPLICATE_TASK);
+        }
```

Comment:
In `MedicationList` and `ConditionList`, duplicates are handled by catching `DuplicateXXException` thrown by the `XXList#add()` method. Your way seems to be a cleaner way to handle it. Would like @BlopApple's input since it involves the features his features as well.

Reply:
I'm fine with either way as long as we standardize the way we handle duplicates by letting `GenericList<T>` to handle it or by letting the `Command` to handle it. 
Will merge this PR for now and request changes in a separate issue.